kern/tc.h

```diff
@@ -131,29 +131,28 @@ static __always_inline int capture_packets(struct __sk_buff *skb, bool is_ingres
     // packet data
     unsigned char *data_start = (void *)(long)skb->data;
     unsigned char *data_end = (void *)(long)skb->data_end;
-    if (data_start + sizeof(struct ethhdr) > data_end) {
+
+    // packet length check
+    if (data_start + sizeof(struct ethhdr) > data_end || 
+        data_start + sizeof(struct ethhdr) + sizeof(struct iphdr) > data_end) 
```

Comment:
`data_start + sizeof(struct ethhdr) > data_end` is unnecessary here. `data_start + sizeof(struct ethhdr) + sizeof(struct iphdr) > data_end` is enough. As we filter ethernet layer info alongside IP layer info at most time.